composable_prepare_nic_configs.yml

```diff
@@ -329,3 +329,41 @@
 
         when: osp_release|int >= 17
 
+      - block:
+          - name: fail if ceph_machine_type or ceph_ifaces are not defined
+            fail:
+              msg: check if ceph_machine_type and ceph_ifaces are defined
+            when: ceph_machine_type is not defined or ceph_ifaces is not defined
+
+          - name: set fact ceph machine type
+            set_fact:
+              ceph_machine_type: "{{ (machine_count[ceph_machine_type]|int >= ceph_node_count|int) | ternary(ceph_machine_type, false) }}"
+
+          - name: fail when insufficient nodes for ceph
+            fail:
+              msg: Not enough nodes of same machine_type for ceph
+            when: ceph_machine_type == false
+
+          - name: set fact for ceph ext iface
+            set_fact:
+              ceph_external_interface: "{{ (ceph_ifaces|length > 1) | ternary(ceph_ifaces[1], ceph_ifaces[0]) }}"
+              ceph_ctlplane_interface: "{{ (ceph_ifaces|length > 1) | ternary(ceph_ifaces[1], ceph_ifaces[0]) }}"
+
+          - name: set fact for ceph storage iface
+            set_fact:
+              ceph_storage_interface: "{{ ceph_ifaces[0] }}"
+            when: ceph_ifaces|length > 1
+
+          - name: prepare ceph-storage.j2.j2
```

Comment:
Nit, the code from lines 333 to 355 appear to be the same as https://github.com/redhat-performance/jetpack/blob/master/composable_prepare_nic_configs.yml#L233-L255. I think it would be better to just move lines 357 to 367 up to that block instead of having a separate block here. It would reduce code duplication.

Reply:
Yes, you're right. I'll get that done.